docs/development/adding-a-package-manager.md

```diff
@@ -111,81 +111,126 @@ Use `updateLockedDependency` to directly update dependencies in lock files.
 
 ## Package files and Lock files
 
-In Renovate terminology, "package files" are the files where human-readable dependency definitions are kept.
-For example, this includes npm's `package.json` file, Maven's `pom.xml` file, and Docker's `Dockerfile`.
+A special Renovate term is "package files".
+Package files have human-readable dependency definitions, for example:
 
-Some package managers may additionally have "lock files", e.g. npm's `package-lock.json`.
-If a lock file is present in a repository then Renovate needs to update both in the same commit, otherwise the update may be "broken".
-Therefore if a new manager is being developed and it is usual to have a lock file, supporting lock file updating should be done from the start.
+- npm's `package.json` file
+- Maven's `pom.xml` file
+- Docker's `Dockerfile`
 
-Supporting lock file updating usually requires Renovate to support a third party tool, e.g. `npm`, `poetry`, etc.
-It's rare and not recommended for Renovate to "reverse engineer" lock file formats and make updates manually instead of calling such tools.
-Adding support for such tools requires adding awareness of each tool to [Containerbase](https://github.com/containerbase/base) first.
+Some package managers may also have "lock files", like npm's `package-lock.json`.
+
+If a repository has a lock file, then Renovate must update the package file _and_ the matching lock file in the same commit.
+Updating both files at once stops Renovate from creating "broken" updates.
+
+### Focus on getting lock file updates working
+
+When you develop a new package manager, which supports lockfiles, focus on getting lock file updates working.
```

Comment:
### What I'm trying to say

> I'm not sure what are you trying to say here. That people should try to avoid to over engineer and rather should directly use CLIs?

This is not about over-engineering, and rather about making sure that a key feature (lockfile updates) is present in the new release of Renovate. At least, that's how I interpret the original text from `@rarkins`. 😄 

#### Silly to release new package support, but leave out lockfile updates

The key here is that _if_ the upstream package manager supports lockfiles, the Renovate developer should make sure that Renovate the program can _update the lockfile_. Its silly to release a new feature for the bot, but then leave out lockfile updates. End-users will complain: "Thanks for the new feature, but why does Renovate not update my lockfiles?"

So that's why I'm saying: Focus on getting lock file updates working.

### My interpretation (feel free to correct me `@rarkins`!)

I think the main point that `@rarkins` is trying to get accross in their developer docs is:

- If the upstream package manager can use lockfiles, make sure that Renovate can update the lockfile

### Difference between original text and my rewrite

The red diff is the original text from `@rarkins`:

```diff
- Therefore if a new manager is being developed and it is usual to have a lock file, supporting lock file updating should be done from the start.
- Supporting lock file updating usually requires Renovate to support a third party tool, e.g. `npm`, `poetry`, etc.
```

And the green diff is how I reworded the original lines:

```diff
+ When you develop a new package manager, which supports lockfiles, focus on getting lock file synchronization working.

+ Supporting lock file synchronization usually means that Renovate runs a third-party tool, like `npm` or `poetry`.
+ The third-party tool then updates the lock file for Renovate.
```

Reply:
I rewrote this to say "focus on getting lock file synchronization working". @secustor does that make more sense?